api/clientset/v1beta1/topology_test.go

```diff
@@ -289,11 +310,59 @@ func TestWatch(t *testing.T) {
 	}
 }
 
-func TestUnstructured(t *testing.T) {
+func TestUpdate(t *testing.T) {
 	cs, _ := setUp(t)
 	tests := []struct {
 		desc    string
 		resp    *http.Response
+		want    *topologyv1.Topology
+		wantErr string
+	}{{
+		desc: "Error",
+		want: &topologyv1.Topology{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Topology",
+				APIVersion: "networkop.co.uk/v1beta1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "doesnotexist",
+				Namespace: "test",
+			},
+		},
+		wantErr: "doesnotexist",
+	}, {
+		desc: "Valid Topology",
+		want: obj1,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			tc := cs.Topology("test")
+			updateObj := tt.want.DeepCopy()
+			updateObj.Spec.Links = append(updateObj.Spec.Links, topologyv1.Link{UID: 1000})
+			update, err := runtime.DefaultUnstructuredConverter.ToUnstructured(updateObj)
+			if err != nil {
+				t.Fatalf("failed to generate update: %v", err)
+			}
+			got, err := tc.Update(context.Background(), &unstructured.Unstructured{Object: update}, metav1.UpdateOptions{})
+			if s := errdiff.Substring(err, tt.wantErr); s != "" {
+				t.Fatalf("unexpected error: %s", s)
+			}
+			if tt.wantErr != "" {
+				return
+			}
+			want := tt.want.DeepCopy()
```

Comment:
why do you need a deep copy and comparison with reflection here instead of cmp/opts?

Reply:
cause cmp wasn't widely used when first did this i will flip it at some point

Reply:
if you won't do a deepcopy, won't you change the `obj` vars in-place as is is a pointer, and risk mutating an object